src/middlewares/auth.js

```diff
@@ -31,36 +43,43 @@ export default (scopes) => {
     // need to get the owner of the token so we can get the appropriate secret.
     const decoded = jwt.decode(token);
 
-    // For now we only allow authenticated requests from the admin user.
-    // When this changes we will have a different secret per sensor and per
-    // user.
-    if (!decoded || !decoded.id || decoded.id !== 'admin') {
+    if (!decoded || !decoded.id || !decoded.scope) {
       return unauthorized(res);
-    };
+    }
 
-    const secret = config.get('adminSessionSecret');
+    if (!validScopes.includes(decoded.scope)) {
+      console.log('Error while authenticating, invalid scope', decoded);
+      return unauthorized(res);
+    }
 
-    // Verify JWT signature.
-    jwt.verify(token, secret, (error, decoded) => {
-      if (error) {
-        return unauthorized(res);
-      }
+    let secretPromise;
+    switch(decoded.scope) {
+      case 'client':
+        secretPromise = db().then(({ Clients }) =>
+          Clients.findById(decoded.id, { attributes: ['secret'] })
+        ).then(client => client.secret);
+        break;
+      case 'user':
+      case 'admin':
+        secretPromise = Promise.resolve(config.get('adminSessionSecret'));
+        break;
+      default:
+        // should not happen because we check this earlier
+        next(new Error(`Unknown scope ${decoded.scope}`));
+    }
 
-      // XXX Get allowed scopes from sensor/user.
+    // Verify JWT signature.
+    secretPromise.then(secret => {
+      jwt.verify(token, secret, (error) => {
+        if (error) {
+          console.log('Error while verifying the token', error);
+          return unauthorized(res);
+        }
 
-      // For now we only allow 'admin' scope.
-      const validScopes = ['admin'].filter(scopeIndex => {
-        return scopes.indexOf(scopeIndex) != -1;
+        req[decoded.scope] = decoded.id;
+        req.authScope = decoded.scope;
```

Comment:
there are 3 locations where we set `req[XXX.scope]` and `req.authScope`: here, and in all `auth/<method>.js`  (basic, facebook, and later google and fxaccounts likely).

That said, maybe we don't want this in `auth/<method>.js` where it's only consumed by `finalizeAuth`; we could simply use req.user or even something else like req.auth.

I'll think of this and maybe we can rework it when implementing google and fx accounts.